### ISSUE TITLE
Display suggestions in context menu

### ISSUE THREAD

**Issue by cameronmcefee** [atom/spell-check#54]
If spell-check detects that a word is misspelled, it'd be nice if the top n suggestions were displayed at the top of the menu when you right click.

Current:
- Click into word
- Right click
- Find "correct spelling"
- Click word

Desired (including https://github.com/atom/spell-check/issues/32)
- Right click word
- Pick suggested spelling

**Comment by a-c-m** [comment]
This please. The current UI is painful/time consuming for those of us who are not so great at speling :)

This is also a fairly established UX pattern used in almost every other editor.

**Comment by jasdel** [comment]
:+1: This please. The `correct spelling` menu is very easy to go unnoticed as a new user.

**Comment by jbjonesjr** [comment]
:+1:

**Comment by lucascosti** [comment]
:thumbsup: Definitely need this. I thought having suggestions in the context menu is already a pretty set paradigm.

**Comment by PaulBeaudet** [comment]
Or just a more easy to learn/use default keybinding to bring up the correction menu would be great

**Comment by pydolan** [comment]
:+1:

**Comment by rugk** [comment]
:+1: Without it that's really not intuitive.

**Comment by beck** [comment]
:+1:

**Comment by cameronmcefee** [comment]
@PaulBeaudet `cmd` + `shift` + `:` is the key command. It took a while but I've built up the muscle memory.

**Comment by orschiro** [comment]
Albeit there is the keyboard shortcut, I would also appreciate support for the context menu. This is such a common convention in software applications to display corrections in the context menu that it is irritating if a software goes against these conventions.

**Comment by bobrocke** [comment]
+1

**Comment by jrawlings** [comment]
:+1:

**Comment by NickRSearcy** [comment]
Anyone have advice for where to start on this? I'd be happy to start working on a pull request

**Comment by kevinsawicki** [comment]
> Anyone have advice for where to start on this? I'd be happy to start working on a pull request

There is an `atom.contextMenu` global with an `add` API, you might want to check out the docs, https://github.com/atom/atom/blob/91bece4336df33a8d84d3c99fc64da50e7027792/src/context-menu-manager.coffee#L111

**Comment by bobrocke** [comment]
@NickRSearcy It looks like many here are rooting for you. How are things going?

**Comment by PhiLhoSoft** [comment]
I approve, although just having "Correct spelling" hidden if right-clicking outside of a misspelled word, and applying to right-clicked word (instead of word under caret) would make me happy.

I saw that the Pigments package is able to do that: right-click on a color, choose _Pigments > Convert to [...]_ and it changes the right-clicked word. It even does that without altering the current selection. Perhaps one can find inspiration in their code...

Note about the suggestion to use `cmd + shift + :` keyboard shortcut: it is a Mac shortcut, it cannot be done on Windows. I tried `Ctrl+Shift+:` and `Ctrl+Alt+Shift+:` without success. Maybe relevant precision: I use a French keyboard (Azerty), although `:` is one of the top-level keys (and `Ctrl+/`, for example, works).

**Comment by dustinring** [comment]
+ 1

FYI-- The current keyboard shortcut `cmd + shift + :` must be done while the cursor is on the incorrectly spelled word. The current UI of check spelling is odd and hard to find the context menu item. I also find it strange that when a line is highlighted the red dots disappear.

**Comment by sylvesterjakubowski** [comment]
:+1:

**Comment by Lior-Knaany-DY** [comment]
+1

**Comment by akshar-patel** [comment]
![image](https://cloud.githubusercontent.com/assets/18311126/14520508/8bc3b664-0242-11e6-9927-d47de55eadd5.png)

**Comment by LukeXF** [comment]
+1

**Comment by JoshuaRogan** [comment]
+1

**Comment by vedyakov** [comment]
+1

**Comment by ground-state** [comment]
+1

**Comment by 50Wliu** [comment]
Shoutout to @speter52 who just implemented this in #172! :tada:
This feature should be available starting in Atom 1.14.0.

**Comment by odd-snail** [comment]
plus 1

**Comment by pluttrell** [comment]
I just checked this out in the Atom Beta for Mac (1.14.0-beta3 x84) and noticed that the suggestions are at the bottom of the context menu. Wonder if they should be moved to the top of the context menu. This would match the UX of most other editors that offer inline spell checking.

**Comment by jbjonesjr** [comment]
@pluttrell noticed that myself, totally agree. :+1:

**Comment by jv2222** [comment]
Any news on moving spelling to the top of the context menu? Seems weird as is.

**Comment by juliaanvaneijndt** [comment]
+1  — would be definitively better

**Comment by Swivelgames** [comment]
👍 I didn't even know there was a keyboard shortcut. That's what I was looking for initially, since I scoured the context menu when I didn't see the suggestions up at the top. Usually the keyboard shortcut is shown to the right of the menu item when there's one attached to the same action.

That's a separate issue, however. I only found the thing because I kept thinking there had to be a way to find the correct spelling if it keeps underlining things.
